packages/fx-core/tests/component/driver/aad/aadAppClient.test.ts

```diff
@@ -312,6 +314,56 @@ describe("AadAppClient", async () => {
         });
     });
 
+    it("should throw error when CredentialInvalidLifetimeAsPerAppPolicy error happens", async () => {
+      const expectedError = {
+        error: {
+          code: "CredentialInvalidLifetimeAsPerAppPolicy",
+        },
+      };
+
+      const mock = new MockAdapter(axiosInstance);
+      mock
+        .onPost(`https://graph.microsoft.com/v1.0/applications/${expectedObjectId}/addPassword`)
+        .reply(400, expectedError);
+
+      await expect(
+        aadAppClient.generateClientSecret(expectedObjectId)
+      ).to.eventually.be.rejected.then((err) => {
+        expect(err instanceof CredentialInvalidLifetimeError).to.be.true;
+        expect(err.source).equals("AadAppClient");
+        expect(err.name).equals("CredentialInvalidLifetime");
+        expect(err.message).equals(
+          "The client secret lifetime exceeds the max value allowed in your tenant. Set a shorter lifetime using the clientSecretExpireDays parameter."
```

Comment:
The client secret lifetime is too long for your tenant. Use a shorter value with the clientSecretExpireDays parameter.